### PR TITLE
Issue 8/organisations

### PR DESCRIPTION
**Pull request by mikeAdamss**
### What is this?

https://app.zenhub.com/workspaces/features-squad-61b72275ac896c0010a1b00b/issues/gss-cogs/reusable-rdf-resources/8 - minus the push to cmd stage which we've pulled out to another task.

### What I've done

As per the card, the only nuance is I've used `org:hasRegisteredSite` for an organisations gov.uk presence and `org:hasPrimarySite` for it's actual website (where they have a separate website).

For example, take ONS:
`org:hasRegisteredSite`:  https://www.gov.uk/government/organisations/office-for-national-statistics
`org:hasPrimarySite`:  https://www.ons.gov.uk

Beyond that I've included what I think is a sensible amount of information, that said there's also probably a data manager task (when we have some) to do some curation (I haven't actually written any of the csv data btw, its all pulled off apis).

I've create the Jenkins job to trigger on commits to main, but only those that alter files in `./pipelines/organisations/*`

Organisations included are those currently on pmd combined with the (non broken) urls here: https://github.com/GSS-Cogs/reusable-rdf-resources/blob/main/organisations.json

Note: I've _not_ included either of these organisations listed in the json:
https://www.gov.uk/api/organisations/the-office-of-gas-and-electricity-markets
https://www.gov.uk/government/organisations/low-carbon-contracts-company-ltd
As these urls 404.

### How to review

Check you're happy with the data in the included csv+csvw and my schema (and therefore the resulting ttl).

An (earlier, less orgs) ttl output can be viewed here: https://ci.ukstats.dev/job/GSS_data/job/reusable-rdf-resources/job/pipelines/job/organisations/20/artifact/out/organisations/organisations.ttl . The Jenkins job works but I cant link it for this as I've toggled it over to use the `main` branch so the build wont work until this is merged.

You can convert the full current csv+csvw to ttl locally with `docker run -v $PWD:/workspace -w /workspace gsscogs/csv2rdf csv2rdf -t ./pipelines/organisationsa/organisations.csv -u ./pipelines/organisations/schema.json -m annotated > organisations.ttl` (from the repo base).

### CODE REVIEW

**Comment by roblinksdata** [comment]
> I've create the Jenkins job to trigger on commits to main, but only those that alter files in ./pipelines/organisations/*

Could we have the job triggering on PRs too so that we can fail a PR if something is wrong (i.e. csv-lint validation fails) but then only have the `publish` step run when in main?

File: pipelines/organisations/organisations.csv [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+https://www.gov.uk/government/organisations/northern-ireland-statistics-and-research-agency,Northern Ireland Statistics and Research Agency,NISRA,"The Northern Ireland Statistics and Research Agency (NISRA), which incorporates the General Register Office (GRO), is an executive agency within the Department of Finance (NI). NISRA is the principal source of official statistics and social research on Northern Ireland. NISRA staff work within a wide range of Northern Ireland Government Departments and Non-Departmental Public Bodies. NISRA is an executive agency, sponsored by the Department of Finance (Northern Ireland) .",https://www.gov.uk/government/organisations/department-of-finance-northern-ireland,https://www.gov.uk/government/organisations/northern-ireland-statistics-and-research-agency,
```
Comment (roblinksdata) [inline pipelines/organisations/organisations.csv]: This list is actually bigger, see [here](https://gss-cogs.github.io/csvcubed-docs/external/guides/linked-data/organisations/) for the full list. I believe I built that up from a query of PMD.

File: pipelines/organisations/schema.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "tables": [
+      {
+          "url": "organisations.csv",
+          "tableSchema": {
+              "columns": [
+                  {
+                      "name": "organisation",
+                      "titles": "organisation",
+                      "datatype": "string",
+                      "suppressOutput": true
```
Comment (roblinksdata) [inline pipelines/organisations/schema.json]: Do we really want to suppress output of the organisation's URI? Surely the `organisation` column is effectively the same as the `hasRegisteredSite` column? Should we just remove it if we're not using it?

File: pipelines/organisations/schema.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                  },
+                  {
+                      "name": "label",
+                      "titles": "label",
+                      "datatype": "string",
+                      "lang": "en",
```
Comment (roblinksdata) [inline pipelines/organisations/schema.json]: Can you stick `"required": true` on `organisation`, `label` and `hasRegisteredSite`?

File: pipelines/organisations/schema.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                      "name": "hasRegisteredSite",
+                      "titles": "hasRegisteredSite",
+                      "datatype": "string",
+                      "propertyUrl": "https://www.gov.uk/government/organisations#hasRegisteredSite"
+                  },
+                  {
+                      "name": "hasPrimarySite",
+                      "titles": "hasPrimarySite",
+                      "datatype": "string",
+                      "propertyUrl": "https://www.gov.uk/government/organisations#hasPrimarySite"
```
Comment (roblinksdata) [inline pipelines/organisations/schema.json]: Are you sure this is the right prefix - shouldn't it be "http://www.w3.org/ns/org#hasPrimarySite"? (And others too)

Comment (mikeAdamss) [reply pipelines/organisations/schema.json]: oops, good spot

File: pipelines/organisations/organisations.csv (hunk 1 above)
Comment (mikeAdamss) [reply pipelines/organisations/organisations.csv]: added via script, comment later.

**Comment by mikeAdamss** [comment]
@robons - off on leave.

I've resolved the minor comments and brought it the extra orgs as defined on gov.uk. They look ok but there's a lot and cursory glance only. I've stuck the script I used on github in case you need it: https://github.com/mikeAdamss/orgs-from-govuk-script.

Haven't touched the Jenkins branches thing at all.

**Review by roblinksdata — APPROVED** [review]
In addition to the above changes, I've create a [multi-branch build pipeline in Jenkins](https://ci.ukstats.dev/job/GSS_data/job/reusable-rdf-resources/job/build-organisations/) so we can have validation of the build status before everything being merged into `main`.

**Pull request merged**